<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from description/youbot_base.xacro  | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="youbot">
  <!-- Define materials -->
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="red">
    <color rgba="1.0 0.0 0.0 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 1.0 0.0 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 1.0 1.0"/>
  </material>
  <material name="yellow">
    <color rgba="1.0 1.0 0.0 1.0"/>
  </material>
  <material name="cyan">
    <color rgba="0.0 1.0 1.0 1.0"/>
  </material>
  <material name="magenta">
    <color rgba="1.0 0.0 1.0 1.0"/>
  </material>
  <material name="gray">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.5 0.0 1.0"/>
  </material>
  <material name="purple">
    <color rgba="0.5 0.0 0.5 1.0"/>
  </material>
  <material name="brown">
    <color rgba="0.6 0.3 0.0 1.0"/>
  </material>
  <!-- <link name="world" />


  <joint name="world_to_base" type="continuous">
    <parent link="world"/>
    <child link="base_footprint"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint> -->
  <!-- 
  <link name="virtual_link_xy">
    <inertial>
      <mass value="0.001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0"
              iyy="0.01" iyz="0.0"
              izz="0.01"/>
    </inertial>
  </link>

  <joint name="virtual_joint_x" type="prismatic">
    <axis xyz="1 0 0"/>
    <parent link="world" />
    <child link="virtual_link_xy" />
    <limit effort="1" lower="-5" upper="5" velocity="1"/>
  </joint>

  <link name="virtual_link_yz">
    <inertial>
      <mass value="0.001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0"
              iyy="0.01" iyz="0.0"
              izz="0.01"/>
    </inertial>
  </link>

  <joint name="virtual_joint_y" type="prismatic">
    <axis xyz="0 1 0"/>
    <parent link="virtual_link_xy" />
    <child link="virtual_link_yz" />
    <limit effort="1" lower="-5" upper="5" velocity="1"/>
  </joint>

  <joint name="virtual_joint_z" type="continuous">
    <axis xyz="0 0 1"/>
    <origin xyz="0 0 0.1"/> 
    <parent link="virtual_link_yz" />
    <child link="base_footprint" />
  </joint> -->
  <link name="base_link">

  </link>
  <link name="base_footprint">
    <!-- <inertial>
      <mass value="0.0001"/>
      <origin rpy="0 0 0" xyz="0 0 1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </collision> -->
  </link>
  <joint name="base_footprint_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <child link="base_link"/>
    <parent link="base_footprint"/>
  </joint>
  <link name="chassis">
    <inertial>
      <mass value="22.0"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="5.652232699207" ixy="-0.009719934438" ixz="1.293988226423" iyy="5.669473158652" iyz="-0.007379583694" izz="3.683196351726"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/youbot_base/base.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/youbot_base/base_convex.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="chassis_joint" type="fixed">
    <parent link="base_link"/>
    <child link="chassis"/>
    <origin xyz="0 0 0"/>
  </joint>
  <joint name="wheel_joint_fl" type="continuous">
    <axis xyz="0 1 0"/>
    <limit effort="30" velocity="10"/>
    <origin rpy="0 0 0" xyz="0.228 0.158 -0.034"/>
    <parent link="base_link"/>
    <dynamics damping="1.0" friction="1.0"/>
    <child link="wheel_link_fl"/>
  </joint>
  <link name="wheel_link_fl">
    <inertial>
      <mass value="0.4"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.012411765597" ixy="-0.000711733678" ixz="0.00050272983" iyy="0.015218160428" iyz="-0.000004273467" izz="0.011763977943"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/youbot_base/front-left_wheel.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <gazebo>
        <surface>
          <friction>
            <ode>
              <mu>1.0</mu>
              <mu2>0.0</mu2>
              <fdir1>1 1 0</fdir1>
              <!-- assuming wheel rolls along x-direction -->
            </ode>
          </friction>
          <bounce>
            <restitution_coefficient>0.0</restitution_coefficient>
            <threshold>1000.0</threshold>
          </bounce>
        </surface>
      </gazebo>
    </collision>
  </link>
  <joint name="wheel_joint_fr" type="continuous">
    <axis xyz="0 -1 0"/>
    <limit effort="30" velocity="10"/>
    <dynamics damping="1.0" friction="1.0"/>
    <origin rpy="3.141592653589793 0 0" xyz="0.228 -0.158 -0.034"/>
    <parent link="base_link"/>
    <child link="wheel_link_fr"/>
  </joint>
  <link name="wheel_link_fr">
    <inertial>
      <mass value="0.4"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.012411765597" ixy="-0.000711733678" ixz="0.00050272983" iyy="0.015218160428" iyz="-0.000004273467" izz="0.011763977943"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/youbot_base/front-right_wheel.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <gazebo>
        <surface>
          <friction>
            <ode>
              <mu>1.0</mu>
              <mu2>0.0</mu2>
              <fdir1>1 -1 0</fdir1>
              <!-- assuming wheel rolls along x-direction -->
            </ode>
          </friction>
          <bounce>
            <restitution_coefficient>0.0</restitution_coefficient>
            <threshold>1000.0</threshold>
          </bounce>
        </surface>
      </gazebo>
    </collision>
  </link>
  <joint name="wheel_joint_bl" type="continuous">
    <axis xyz="0 1 0"/>
    <limit effort="30" velocity="10"/>
    <dynamics damping="1.0" friction="1.0"/>
    <origin rpy="0 0 0" xyz="-0.228 0.158 -0.034"/>
    <parent link="base_link"/>
    <child link="wheel_link_bl"/>
  </joint>
  <link name="wheel_link_bl">
    <inertial>
      <mass value="0.4"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.012411765597" ixy="-0.000711733678" ixz="0.00050272983" iyy="0.015218160428" iyz="-0.000004273467" izz="0.011763977943"/>
    </inertial>
    <visual>
      <origin rpy="3.141592653589793 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/youbot_base/back-left_wheel.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <gazebo>
        <surface>
          <friction>
            <ode>
              <mu>1.0</mu>
              <mu2>0.0</mu2>
              <fdir1>-1 1 0</fdir1>
              <!-- assuming wheel rolls along x-direction -->
            </ode>
          </friction>
          <bounce>
            <restitution_coefficient>0.0</restitution_coefficient>
            <threshold>1000.0</threshold>
          </bounce>
        </surface>
      </gazebo>
    </collision>
  </link>
  <joint name="wheel_joint_br" type="continuous">
    <axis xyz="0 -1 0"/>
    <limit effort="30" velocity="10"/>
    <dynamics damping="1.0" friction="1.0"/>
    <origin rpy="3.141592653589793 0 0" xyz="-0.228 -0.158 -0.034"/>
    <parent link="base_link"/>
    <child link="wheel_link_br"/>
  </joint>
  <link name="wheel_link_br">
    <inertial>
      <mass value="0.4"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.012411765597" ixy="-0.000711733678" ixz="0.00050272983" iyy="0.015218160428" iyz="-0.000004273467" izz="0.011763977943"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/youbot_base/back-right_wheel.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <gazebo>
        <surface>
          <friction>
            <ode>
              <mu>1.0</mu>
              <mu2>0.0</mu2>
              <fdir1>-1 -1 0</fdir1>
              <!-- assuming wheel rolls along x-direction -->
            </ode>
          </friction>
          <bounce>
            <restitution_coefficient>0.0</restitution_coefficient>
            <threshold>1000.0</threshold>
          </bounce>
        </surface>
      </gazebo>
    </collision>
  </link>
  <joint name="base_laser_front_hokuyo_urg04_laser_joint" type="fixed">
    <axis xyz="0 1 0"/>
    <origin rpy="3.141592653589793 3.141592653589793 3.141592653589793" xyz="0.3 0 -0.03"/>
    <parent link="base_link"/>
    <child link="base_laser_front_link"/>
  </joint>
  <link name="base_laser_front_link" type="laser">
    <inertial>
      <mass value="0.16"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/sensors/hokuyo.dae"/>
      </geometry>
      <material name="black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/nbf2/youbot_ws/install/youbot/share/youbot/meshes/sensors/hokuyo_convex.dae"/>
      </geometry>
    </collision>
  </link>
  <!-- 
  <transmission name="wheel_trans_fl">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_fl"/>
    <actuator name="wheel_motor_fl">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_trans_fr">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_fr"/>
    <actuator name="wheel_motor_fr">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_trans_bl">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_bl"/>
    <actuator name="wheel_motor_bl">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_trans_br">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_br"/>
    <actuator name="wheel_motor_br">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission> -->
  <transmission name="wheel_trans_fl">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_fl">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_motor_fl">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_trans_fr">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_fr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_motor_fr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_trans_bl">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_bl">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_motor_bl">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_trans_br">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_joint_br">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_motor_br">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>79.2380952381</mechanicalReduction>
    </actuator>
  </transmission>
</robot>
